### PR TITLE
Fix issue #1174 (division by zero in gradient computation)

### PR DESCRIPTION
**Pull request by ibusko**

### CODE REVIEW

**Comment by ibusko** [comment]
The CI tests that get cancelled seem to be running into a timeout. Most of them stop when running `test_ellipse`. When I run `tox` on my computer I can see that they eventually pass. I believe this is related to issue #1168. Not only `build_model`, but the core sampling algorithm is `fit_ellipse` should be made faster by using Cython.

**Comment by larrybradley** [comment]
It looks like the tests failed to due an internet timeout.  I've restarted them.  Let's see what happens.

**Review by larrybradley — APPROVED** [review]
Thanks, @ibusko.

File: CHANGES.rst [hunk 1]
```diff
@@ -79,6 +79,11 @@ Bug Fixes
 
   - Fixed ``SourceProperties`` Kron mask correction. [#1167]
 
+- ``photutils.isophote``
+
+  - Fix computation of gradient relative error when gradient=0
```
Comment (larrybradley) [inline CHANGES.rst:84]:
```suggestion
  - Fix computation of gradient relative error when gradient=0. [#1180]
```

**Review by larrybradley — CHANGES_REQUESTED** [review]

**Review by larrybradley — APPROVED** [review]

**Comment by larrybradley** [comment]
@ibusko This is a real test failure.  I can reproduce the test failure locally.  The failed test is in `test_fitter.py` in the `TestM51.test_m51()` method.  This test loads the remote `M51.fits` image, so to run it locally add the `--remote-data=any` flag:  `pytest --remote-data-any`.  Specifically, `isophote.intens` is now returning 155.046967 when 156.3 is expected:  https://github.com/astropy/photutils/pull/1180/checks?check_run_id=2115521635#step:7:276

**Review by larrybradley — APPROVED** [review]
Thanks, @ibusko!

**Pull request merged**